### PR TITLE
[Doc] update length of names in system limit

### PR DESCRIPTION
**Pull request by evelynzhaojie**
Why I'm doing: the length limit of some names in SR is inconsistent with code

What I'm doing: update the name length limit

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/zh/reference/System_limit.md [hunk 1]
```diff
@@ -8,13 +8,15 @@ displayed_sidebar: "Chinese"
 
 - StarRocks 采用 MySQL 协议进行通信，用户可通过 MySQL Client 或者 JDBC 连接到 StarRocks 集群。选择 MySQL Client 版本时建议采用 5.1 及之后的版本，5.1 之前版本不支持长度超过 16 个字符的用户名。
 
-- 对数据目录名 (Catalog)、数据库名、表名、视图名、分区名、列名、用户名、角色名的要求：
+- 对名称的要求，包括数据目录名 (Catalog)、数据库名、表名、视图名、分区名、列名、用户名、角色名等：
   - 只能由数字 (0-9)、字母 (a-z或A-Z)，以及下划线 (\_) 组成。**用户名 (username) 可以使用纯数字命名。**
-  - 长度不能超过 64 个字符。**其中，数据目录名、数据库名、表名、列名不能超过 1023 字符 (≤ 1023)。**
-  - 数据目录名、数据库名、表名、视图名、分区名、以及角色名均只能以小写或大写字母开头。
```
Comment (evelynzhaojie) [inline docs/zh/reference/System_limit.md:14]: 之前写的是只能以字母开头，但是实际试了一下，下划线也都可以

**Review by Astralidea — APPROVED** [review]

File: docs/zh/reference/System_limit.md (hunk 1 above)
Comment (EsoragotoSpirit) [reply docs/zh/reference/System_limit.md:14]: 只有列名的 `__op` 和 `__row` （两个连续下滑线）不行，单个下划线都行的。其他名称没有这个限制

Comment (EsoragotoSpirit) [reply docs/zh/reference/System_limit.md:14]: 但是可以用fe 参数开启

**Review by EsoragotoSpirit — APPROVED** [review]

**Comment by evelynzhaojie** [comment]
@mergify backport branch-3.1

**Pull request merged**